Content.Client/Fluids/PuddleVisualsSystem.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Linq;
+using Content.Shared.Fluids;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Shared.Random;
+
+namespace Content.Client.Fluids
+{
+    [UsedImplicitly]
+    public sealed class PuddleVisualsSystem : VisualizerSystem<PuddleVisualsComponent>
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<PuddleVisualsComponent, ComponentInit>(OnComponentInit);
+        }
+
+        private void OnComponentInit(EntityUid uid, PuddleVisualsComponent puddleVisuals, ComponentInit args)
+        {
+            if (!TryComp(uid, out AppearanceComponent? appearance))
+            {
+                return;
+            }
+
+            if (!TryComp(uid, out SpriteComponent? sprite))
+            {
+                return;
+            }
+
+            if (!appearance.TryGetData(PuddleVisuals.VisualSeed, out float visualSeed))
+            {
+                return;
+            }
+
+            var maxStates = sprite.BaseRSI?.ToArray();
+
+            if (maxStates is not { Length: > 0 }) return;
+
+            int selectedState = (int) Math.Floor(visualSeed * maxStates.Length); // uses the visualSeed to randomly select an index for which RSI state to use.
+            sprite.LayerSetState(PuddleVisualLayers.Puddle, maxStates[selectedState].StateId); // sets the sprite's state via our randomly selected index.
+
+            int rotationDegrees = (int) Math.Floor(visualSeed * 360); // uses the visualSeed to randomly select a rotation for our puddle sprite.
+            sprite.Rotation = Angle.FromDegrees(rotationDegrees); // sets the sprite's rotation to the one we randomly selected.
```

Comment:
using the same random number for both state and rotation seems a bit icky, but you should at least do so something like:
```cs
float rotationDegrees = ( visualSeed * maxStates.Length - selectedState ) * 360;
```
because currently, if you had 10 states, the zeroth state (visual seed between 0 and 0.1) will only ever have a rotation between (0-36). This at least ensures that it can have all angles.

Reply:
Ah, you're quite right. Thank you for the example, this was very clear.

Should I just use two different seeds then? I don't mind doing it that way if that's your preference.

Reply:
Using one random number works fine, and is technically means there is ever so slightly less data the sever has to send to clients... but it still feels weird? I'm not really sure which is better.

Unless anyone else feels strongly, I guess just do whatever you prefer

Reply:
Thanks! I've replaced my formula with yours. Gonna keep it simple with one `visualSeed`.

Quick question though @ElectroJr - why do we subtract the `selectedState` here?